src/server/hitbox.ts

```diff
@@ -0,0 +1,142 @@
+export class Hitbox {
+    private _x: number;
+    private _y: number;
+
+    constructor (x: number, y: number) {
+        this._x = x; 
+        this._y = y;
+    }
+
+    get x () : number {
+        return this._x
+    }
+
+    get y () : number {
+        return this._y;
+    }
+
+    // Absolute position update
+    updatePos(x: number, y: number) {
+        this._x = x;
+        this._y = y;
+    }
+    // Relative position update
+    move(dx: number, dy: number) {
+        this._x += dx;
+        this._y += dy;
+    }
+
+    // Test if the passed hitbox will collide if it moves by the given velocity
+    testMovement (hitbox: Hitbox, dx: number, dy: number) : boolean {
+        hitbox.move(dx, dy);
+        let result = this.hasCollision(hitbox);
+        hitbox.move(-dx, -dy);
+        return result;
+    }
+
+    // Checks if a collisions occurs
+    hasCollision (hitbox: Hitbox) : boolean {return false}; // Should be an abstract method
```

Comment:
```suggestion
    abstract hasCollision (hitbox: Hitbox) : boolean;
```

Reply:
@Naidah I was thinking of doing this but it seems typescript doesn't like abstract methods in non-abstract class. Let me know if I'm wrong.